packages/react/src/components/Geo/LocationSearch/index.tsx

```diff
@@ -15,8 +19,31 @@ const LOCATION_SEARCH_OPTIONS = {
 
 const LOCATION_SEARCH_CONTAINER = 'geocoder-container';
 
+interface LocationSearchProps extends MaplibreGeocoderOptions {
+  /**
+   * Emitted when the input is cleared
+   */
+  onClear?: () => void;
+  /**
+   * Emitted when the geocoder is looking up a query
+   */
+  onLoading?: (query: string) => void;
+  /**
+   * Fired when the geocoder returns a response
+   */
+  onResults?: (results: MaplibreGeocoderResults) => void;
+  /**
+   * Fired when input is set
+   */
+  onResult?: (result: MaplibreGeocoderResult) => void;
+}
+
+type EventType = 'clear' | 'loading' | 'result' | 'results';
+
 type AmplifyLocationSearch = IControl & {
   addTo: (container: string) => void;
+  on: (eventType: EventType, callback: (event) => void) => void;
+  off: (eventType: EventType, callback: (event) => void) => void;
```

Comment:
@sreeramsama Can we help out by adding the event types for the callbacks here?

Reply:
Added type in latest commit